tinygrad/codegen/linearizer.py

```diff
@@ -44,10 +44,11 @@ def uop_alu_idx(self, a:UOp, b, ops, ctx:Linearizer, op, dtype:DType=dtypes.int3
     return self.uops.add(UOps.ALU, dtype, (a, render_b), op)
 
   # NOTE: the consts have to be cached for deduping of downstream uops to work
-  def const(self, b:ConstType, dtype:DType=dtypes.int32, insert_before=None) -> UOp:
+  def const(self, b:ConstType, dtype:DType=dtypes.int32, insert_before:Optional[UOp|int]=None) -> UOp:
     return self.uops.add(UOps.CONST, dtype, tuple(), b, insert_before=insert_before)
 
-  def cast(self, val: UOp, dtype) -> UOp: return self.uops.add(UOps.CAST, dtype, (val,)) if val.dtype != dtype else val
+  def cast(self, val: UOp, dtype:DType, insert_before:Optional[UOp|int]=None) -> UOp:
```

Comment:
@Qazalin a little unrelated but is this function dead code? I deleted it and ran some tests and everything still worked

Reply:
yeah.. i think so

Reply:
I just used it instead, it makes a lot of the casts cleaner but I can use `self.uops.add` if it's better

Reply:
i think self.uops.add is better for this diff - refactoring casts can be a different diff if you want